client/src/views/NodeDetail.vue

```diff
@@ -49,54 +49,70 @@
   </v-container>
 </template>
 
-<script>
-import api from '@/api'
+<script lang="ts">
+import Vue from 'vue';
 
-export default {
+import api from '@/api';
+import { KeyValue } from '@/types';
+
+interface EdgeRecord {
+  id: string;
+  from: string;
+  to: string;
+}
+
+interface Connection {
+  id: string;
+  airport: string;
+}
```

Comment:
Mainly because I don't know what I'm doing yet :smile_cat: 

I believe it's best to put types that have semantics on the project scale in `types.ts`, and auxiliary types that are local to some set of functions, that don't necessarily mean anything important, alongside those functions. `Connection` maybe belongs where it is, since it's being used to type two instances of a function argument without a larger use of the Connection concept in the codebase (besides the fact that whatever else, the type should probably not have a field named `airport`).

There's a further issue of what to put in a given project's `types.d.ts` file for consumption in downstream projects, but that one is likely moot for us, since we're developing an end-user application, rather than a library, etc.

Reply:
Ok, seems like a judgement thing. There should probably be some guidance for this going forward.

And yeah, `airport` should not be there, can that be easily removed in this PR or is that a separate issue?

Reply:
> Ok, seems like a judgement thing. There should probably be some guidance for this going forward.
> 

Opened #181 to address this.

> And yeah, `airport` should not be there, can that be easily removed in this PR or is that a separate issue?

I'd prefer not to fix it on this branch, since it's simply an artifact of the first exemplar dataset we ever used, and it doesn't affect the correct operation of the codebase. I opened #182 to cover this.

Reply:
Works for me.

Reply:
I will actually move my own type declarations to types.ts as appropriate before resolving this, just FYI.